src/draw/DefaultDrawer.js

```diff
@@ -106,466 +160,464 @@ class DefaultDrawer {
-            .type;
-
-          if (!this.actions.link.targetList.includes(targetType)) {
-            this.__unsetAddLink();
-          }
-        },
-        { once: true },
-      );
-      this.__addLink(allowedTarget);
-    });
+    return true;
   }
 
   /**
-   * Delete component
-   * @private
-   */
-  __deleteComponent(componentId) {
-    if (this.events.DeleteEvent) {
-      this.events.DeleteEvent.next({
-        id: componentId,
-      });
+   *
+   * @param {PointerEvent} event - The click event
+   * @param {Node} data - The clicked component data
+   */
+  clickHandler(event, data) {
+    event.stopPropagation();
+    this.__selectComponent(data.data.id);
+  }
+
+  drawComponents() {
+    this.shadowRoot = { children: this.pluginData.components, id: '__shadowRoot', name: '' };
+    const groupedNodes = this.buildTree();
+    const clicked = this.clickHandler.bind(this);
+
+    const drag = this.setupDragBehavior();
+
+    const node = this.svg
+      .select('.container')
+      .selectAll('g')
+      .data(groupedNodes, (data) => data)
+      .join('g')
+      .attr('id', (data) => data[0])
+      .selectAll('g')
+      .data((data) => data[1])
+      .join('g')
+      .attr('id', (data) => data.data.id)
+      .on('click', clicked)
+      .call(drag)
+      .attr('transform', (data) => `translate(${data.x0},${data.y0})`);
+
+    node
+      .filter((data) => data.data.id !== '__shadowRoot')
+      .attr('class', (data) => `component
+        component-${data.data.definition.model}
+        ${data.data.definition.type}`)
+      .html((data) => this.resources.models[data.data.definition.model])
+      .select('svg')
+      .attr('id', (data) => `svg-${data.data.id}`)
+
+      .attr('height', (data) => data.y1 - data.y0)
+      .attr('width', (data) => data.x1 - data.x0);
+
+    node.select('.component-name')
+      .text((data) => data.data.name);
+    node.select('.component-type')
+      .text((data) => data.data.definition.type);
+    node.select('.component-icon')
+      .html((data) => this.resources.icons[data.data.definition.icon]);
+
+    node.select('rect')
+      .filter((d) => d.data?.definition?.isContainer)
+      .attr('height', (data) => data.y1 - data.y0)
+      .attr('width', (data) => data.x1 - data.x0);
+
+    node.select('.component-container')
+      .attr('height', (data) => Math.max(
+        data.y1
+            - (data.y0 + this.minHeight)
+            - this.margin,
+        this.minHeight,
+      ))
+      .attr('width', (data) => Math.max(data.x1 - data.x0 - 2 * this.margin, this.minWidth))
+      .attr('x', () => this.margin)
+      .attr('y', () => this.minHeight)
+      .filter((data) => data.children)
+      .append((data) => d3.select(`#group-${data.data.id}`).node());
+
+    node.select('.component-container>rect').attr('data-parentId', (d) => d.data.id);
+  }
+
+  /**
+   * Build d3 hierarchy and treemap layout.
+   * @return {[[groupId, [Node]]]} - The nodes grouped by parent.
```

Comment:
I think the type is not good for the documentation can you please check with the builded docs

Reply:
Tested, types work fine.